src/shared/sam/cli/samCliContext.ts

```diff
@@ -45,6 +46,13 @@ export function getSamCliContext() {
     return samCliContext
 }
 
+export async function getSamCliVersion(context: SamCliContext): Promise<string> {
+    const result = await context.validator.detectValidSamCli()
```

Comment:
I'm guessing that you're using the validator because it happens to cache and store the sam cli version. Maybe that is something that should be pulled out of validator and available standalone. If you agree, consider writing a ticket, and adding a TODO comment here linking the ticket

Reply:
I feel like the idea behind this function was that we wouldn't have to check for specific versions ever and that any discrepancies in functionality would be dealt with by bumping the min version. I'm wondering, could we instead bump the min version to avoid this additional code?

I agree that we may want to move the actual version getter to a different function if we want to keep using the older versions, though.